src/opentimelineio/composition.cpp

```diff
@@ -498,4 +499,18 @@ int64_t Composition::_bisect_left(
     return *lower_search_bound;
 }
 
+bool Composition::has_clips() const {
+    for (auto child: children()) {
```

Comment:
I recently added some functions for working with child objects, you could also write this as:
```
ErrorStatus error_status;
return children_if<Clip>(&error_status).size() > 0;
```
```

Reply:
That's nice! I noticed there were some new changes to the cpp when I rebased. I don't suppose there's a way to short-circuit the children_if call so it stops on the first match? That would be the advantage of keeping this as-is, but I admit it's not nice to have duplicate functionality.

Reply:
Good point, the short-circuit is nice to have.

Reply:
I could perhaps add a `children_has<Clip>` to fit more with the existing pattern, that would do the short circuit.

Reply:
The template seems nice, that way you could also check for the existence of any transitions, etc. I wonder if it would be useful to return the instance instead of a bool?

Reply:
Sure.  I think it makes sense to do something like `optional<T> children_first<T>` that would return the first instance of T found.  In my case I simply need to check whether the optional is empty.

Reply:
I guess I was thinking of something more like:
```
template<typename T>
Retainer<T> child_if(
    ErrorStatus* error_status,
    optional<TimeRange> search_range = nullopt) const;
```
To match the ```children_if()``` function; that way you could just write:
```
ErrorStatus error_status;
return child_if<Clip>(&error_status);
```
If this seems like too much for this PR I would also be happy to add it as a separate PR since I implemented the other functions.

Reply:
I'm good with that too.  Because of the CMake issues on Windows, I'm going to have to put it on hold for a bit until I can fix that. If you do it in the meantime, I'll update and use it.  If not, I'll do it as part of this PR.  So essentially I'll leave it up to you :)

Reply:
I created a draft PR for the child_if() functions:
https://github.com/PixarAnimationStudios/OpenTimelineIO/pull/1002

By making it a general function and supporting the various otio container types it turned out to be more code changes than I was hoping for...